### PR TITLE
Add new data source cen_instances

### PR DESCRIPTION
**Pull request by GiantRice**
The CENs data source lists a number of CENs resource information owned by an Alicloud account.

The results of acceptance tests are as below:
$TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudCenInstancesDataSource -timeout=300m
=== RUN   TestAccAlicloudCenInstancesDataSource_cen_id
--- PASS: TestAccAlicloudCenInstancesDataSource_cen_id (7.64s)
=== RUN   TestAccAlicloudCenInstancesDataSource_name_regex
--- PASS: TestAccAlicloudCenInstancesDataSource_name_regex (8.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	16.415s

### CODE REVIEW

File: alicloud/data_source_alicloud_cen_instances.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		Read: dataSourceAlicloudCenInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"cen_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				MinItems: 1,
+				MaxItems: 5,
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: Please use `ids` instead '`cen_ids`, and I think it should not be limited in 1-5.

File: alicloud/data_source_alicloud_cen_instances.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+				ForceNew:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"cen_bandwidth_package_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				MinItems: 1,
+				MaxItems: 5,
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: Please use `bandwidth_package_ids ` instead `cen_bandwidth_package_ids `.

File: alicloud/data_source_alicloud_cen_instances.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+			},
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			// Computed values
+			"cens": {
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: Please use `instances` instead `cens`

File: alicloud/data_source_alicloud_cen_instances.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cen_bandwidth_package_ids": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: Please use `bandwidth_package_ids` instead `cen_bandwidth_package_ids.`

File: alicloud/data_source_alicloud_cen_instances.go [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+						"instance_ids": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: Please use `child_instance_ids` instead `instance_ids`.

File: alicloud/data_source_alicloud_cen_instances.go [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+				instanceIds = append(instanceIds, inst.ChildInstanceId)
+			}
+
+		}
+
+		if len(resp.ChildInstances.ChildInstance) < PageSizeLarge {
+			break
+		}
+
+		childInstancesReq.PageNumber += requests.NewInteger(1)
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances.go]: There is a bug in line 232 and you can refer to https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/data_source_alicloud_vpcs.go#L132

File: alicloud/data_source_alicloud_cen_instances_test.go [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+const testAccCheckAlicloudCenInstancesDataSourceCenNameRegexConfig = `
+resource "alicloud_cen_instance" "tf-testAccCen" {
+	name = "tf-testAccCenConfig"
+	description = "tf-testAccCenConfigDescription"
+}
+
+data "alicloud_cen_instances" "tf-testAccCen" {
+	name_regex = "${alicloud_cen_instance.tf-testAccCen.name}"
+}
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances_test.go]: Please add another testcase that support using 'bandwidth_package_ids' to fetch instances.

File: alicloud/data_source_alicloud_cen_instances_test.go [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+				Config: testAccCheckAlicloudCenInstancesDataSourceCenNameRegexConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_cen_instances.tf-testAccCen"),
+					resource.TestCheckResourceAttr("data.alicloud_cen_instances.tf-testAccCen", "cens.0.name", "tf-testAccCenConfig"),
+					resource.TestCheckResourceAttr("data.alicloud_cen_instances.tf-testAccCen", "cens.0.description", "tf-testAccCenConfigDescription"),
+				),
+			},
+		},
+	})
+}
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cen_instances_test.go]: Please check all of resource attributes in each test case.

File: website/docs/d/cen_instances.html.markdown [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - ID of the CEN.
+* `name` - Name of the CEN.
+* `status` - Status of the CEN, including "Creating", "Active" and "Deleting".
+* `cen_bandwidth_package_ids` - List of CEN Bandwidth Package IDs in the specified CEN.
+* `instance_ids` - List of child instance IDs in the specified CEN.
+* `description` - Description of the CEN.
```
Comment (xiaozhu36) [inline website/docs/d/cen_instances.html.markdown]: Please refer to other datasource export format and improve this datasource's  `Attributes Reference`

**Comment by GiantRice** [comment]
> The CENs data source lists a number of CENs resource information owned by an Alicloud account.
> 
> The results of acceptance tests are as below:
> $TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudCenInstancesDataSource -timeout=300m
> === RUN TestAccAlicloudCenInstancesDataSource_cen_id
> --- PASS: TestAccAlicloudCenInstancesDataSource_cen_id (7.64s)
> === RUN TestAccAlicloudCenInstancesDataSource_name_regex
> --- PASS: TestAccAlicloudCenInstancesDataSource_name_regex (8.71s)
> PASS
> ok github.com/terraform-providers/terraform-provider-alicloud/alicloud	16.415s

TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudCenInstancesDataSource_ -timeout=300m
=== RUN   TestAccAlicloudCenInstancesDataSource_cen_id
--- PASS: TestAccAlicloudCenInstancesDataSource_cen_id (9.71s)
=== RUN   TestAccAlicloudCenInstancesDataSource_name_regex
--- PASS: TestAccAlicloudCenInstancesDataSource_name_regex (8.39s)
=== RUN   TestAccAlicloudCenInstancesDataSource_multi_cen_ids
--- PASS: TestAccAlicloudCenInstancesDataSource_multi_cen_ids (29.42s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	47.570s

**Pull request merged**